packages/components/src/core/InputCheckbox/styles.ts

```diff
@@ -64,6 +68,16 @@ export const StyledCheckbox = styled(RawCheckbox)`
         }
       }
 
+      &.MuiCheckbox-root {
+        &.Mui-focusVisible {
+          outline: 5px auto Highlight;
+          outline: 5px auto -webkit-focus-ring-color;
```

Comment:
I think these outlines would be nice to have somewhere reusable.  I ran into trouble trying to make them into a mixin.

Reply:
Wow nice focus-visible only style 🔥 !

I haven't tested this yet, but maybe this reusable function would work?

```ts
function focusVisibleA11yStyle(props: CommonThemeProps) {
  const corners = getCorners(props);
  const spaces = getSpaces(props);

  return `
    &.MuiCheckbox-root {
      &.Mui-focusVisible {
        outline: 5px auto Highlight;
        outline: 5px auto -webkit-focus-ring-color;
      }
      border-radius: ${corners?.s}px;
      padding: ${spaces?.xxs}px;
      margin: 3px;
    }
  `;
}

export const StyledCheckbox = styled(RawCheckbox)`
  ${focusVisibleA11yStyle}  // <--------- invoke it here!

  ${(props) => {
    const colors = getColors(props);
    const iconSizes = getIconSizes(props);

    return `
      color: ${colors?.gray[400]};
      &:hover {
        color: ${colors?.gray[500]};
        background-color: transparent;
      }
      &.Mui-disabled {
        color: ${colors?.gray[300]};
      }
      &.Mui-checked {
        color: ${colors?.primary[400]};
        &:hover {
          color: ${colors?.primary[500]};
          background-color: transparent;
        }
        &.Mui-disabled {
          color: ${colors?.primary[300]}
        }
      }

      .MuiSvgIcon-root {
        height: ${iconSizes?.input.height}px;
        width: ${iconSizes?.input.width}px;
      }
    `;
  }}
`;
```

Reply:
I created a function in mixins/a11y.ts
```
export function focusVisibleA11yStyle() {
  return `
      &.Mui-focusVisible, &:focus-visible {
        outline: 5px auto Highlight;
        outline: 5px auto -webkit-focus-ring-color;
      }
  `;
}
```
And used it in the component
```
export const StyledCheckbox = styled(RawCheckbox)`
  ${(props) => {
    const colors = getColors(props);
    const corners = getCorners(props);
    const iconSizes = getIconSizes(props);
    const spaces = getSpaces(props);
    return `
      color: ${colors?.gray[400]};
      &:hover {
        color: ${colors?.gray[500]};
        background-color: transparent;
      }
      &.Mui-disabled {
        color: ${colors?.gray[300]};
      }
      &.Mui-checked {
        color: ${colors?.primary[400]};
        &:hover {
          color: ${colors?.primary[500]};
          background-color: transparent;
        }
        &.Mui-disabled {
          color: ${colors?.primary[300]}
        }
      }

      &.MuiCheckbox-root {
        ${focusVisibleA11yStyle()}
        border-radius: ${corners?.s}px;        
        padding: ${spaces?.xxs}px;
        margin: 3px;
      }

      .MuiSvgIcon-root {
        height: ${iconSizes?.input.height}px;
        width: ${iconSizes?.input.width}px;
      }
    `;
  }}
`;
```
I'm still having a small issue where the function needs to be invoked in order to work.  Same in the button:
```

    return `
      ${outlineBorder}
      ${focusVisibleA11yStyle()}
      padding: ${padding};
      min-width: 120px;
      height: 34px;
      &:hover {
        color: white;
        background-color: ${colors?.primary[500]};
        box-shadow: none;
      }
      &:active {
        color: white;
        background-color: ${colors?.primary[600]};
        box-shadow: none;
      }
      &:disabled {
        color: ${colors?.gray[400]};
        background-color: ${colors?.gray[300]};
        border-color: ${colors?.gray[300]};
      }
      .MuiButton-startIcon {
        margin-right: ${spacings?.m}px;
      }
      .MuiButton-endIcon {
        margin-left: ${spacings?.m}px;
      }
    `;
  }}
`;
```

Reply:
I created a function and using it as `${focusVisibleA11yStyle}` in the button component works, but in the checkbox i have to invoke it with `${focusVisibleA11yStyle()}`.  I'm not seeing why these would be different.

Function
```
export function focusVisibleA11yStyle() {
  return `
      &.Mui-focusVisible, &:focus-visible {
        outline: 5px auto Highlight;
        outline: 5px auto -webkit-focus-ring-color;
      }
  `;
}
```

Checkbox
```
export const StyledCheckbox = styled(RawCheckbox)`
  ${(props) => {
    const colors = getColors(props);
    const corners = getCorners(props);
    const iconSizes = getIconSizes(props);
    const spaces = getSpaces(props);
    return `
      color: ${colors?.gray[400]};
      &:hover {
        color: ${colors?.gray[500]};
        background-color: transparent;
      }
      &.Mui-disabled {
        color: ${colors?.gray[300]};
      }
      &.Mui-checked {
        color: ${colors?.primary[400]};
        &:hover {
          color: ${colors?.primary[500]};
          background-color: transparent;
        }
        &.Mui-disabled {
          color: ${colors?.primary[300]}
        }
      }

      &.MuiCheckbox-root {
        ${focusVisibleA11yStyle()}
        border-radius: ${corners?.s}px;        
        padding: ${spaces?.xxs}px;
        margin: 3px;
      }

      .MuiSvgIcon-root {
        height: ${iconSizes?.input.height}px;
        width: ${iconSizes?.input.width}px;
      }
    `;
  }}
`;
```
Button
```
const ButtonBase = styled(Button, {
  shouldForwardProp: (prop) => !sdsPropNames.includes(prop as string),
})`
  box-shadow: none;
  ${(props) => {
    const { variant } = props;
    const colors = getColors(props);
    const spacings = getSpaces(props);

    const containedPadding = `${spacings?.xs}px ${spacings?.l}px`;

    // (thuang): outline variant has border 1px, so padding needs to be smaller
    const outlinedPadding = `${(spacings?.xs || 0) - 1}px ${
      (spacings?.l || 0) - 1
    }px`;

    const padding = variant === "outlined" ? outlinedPadding : containedPadding;
    const outlineBorder =
      variant === "outlined" ? `border-color: ${colors?.primary[400]};` : "";

    return `
      ${outlineBorder}
      ${focusVisibleA11yStyle}
      padding: ${padding};
      min-width: 120px;
      height: 34px;
      &:hover {
        color: white;
        background-color: ${colors?.primary[500]};
        box-shadow: none;
      }
      &:active {
        color: white;
        background-color: ${colors?.primary[600]};
        box-shadow: none;
      }
      &:disabled {
        color: ${colors?.gray[400]};
        background-color: ${colors?.gray[300]};
        border-color: ${colors?.gray[300]};
      }
      .MuiButton-startIcon {
        margin-right: ${spacings?.m}px;
      }
      .MuiButton-endIcon {
        margin-left: ${spacings?.m}px;
      }
    `;
  }}
`;
```

Reply:
Ah yes, so it depends on where `focusVisibleA11yStyle` is placed!

If we put `focusVisibleA11yStyle` inside `${(props) => { ... }}`, we'll need to manually invoke `focusVisibleA11yStyle`, like so: `${focusVisibleA11yStyle()}`, since Emotion already called the parent level function. However, if we place `focusVisibleA11yStyle` at the root styled component level, then Emotion will invoke the function for us!

E.g.,

```ts
// Root level
const Foo = styled.div`
  {focusVisibleA11yStyle}

  ...other styles
`

// Nesting focusVisibleA11yStyle inside `${(props) => {...}}

const Foo = styled.div`
  ${(props) => {
    return `
      ${focusVisibleA11yStyle(props)};

      ... other styles
    `
  }
`
```

Hope this helps!

Reply:
For Button, I think we can lift the `focusVisibleA11yStyle` outside of `${(props) => { ... }}`, since it doesn't depend on anything inside that function anyway 😄 ?

Like so:

```ts
const ButtonBase = styled(Button, {
  shouldForwardProp: (prop) => !sdsPropNames.includes(prop as string),
})`
  box-shadow: none;

  ${focusVisibleA11yStyle} // <--- HERE!

  ${(props) => {
    const { variant } = props;
    const colors = getColors(props);
    const spacings = getSpaces(props);
    ...
  }
```